### ISSUE TITLE
BBC: Blead Breaks Code::TidyAll

### ISSUE THREAD

**Issue by cjg-cguevara** [Perl/perl5#21639]
This is a bug report for perl from "Carlos Guevara" <carlos@carlosguevara.com>,
generated with the help of perlbug 1.43 running under perl 5.39.5.

-----------------------------------------------------------------

BBC: Blead Breaks Code::TidyAll

Please see http://fast-matrix.cpantesters.org/?dist=Code::TidyAll

---
**Flags**
- category=core
- severity=low
---
**Perl configuration**
```
Site configuration information for perl 5.39.5:

Configured by cpan at Wed Nov 15 21:38:16 EST 2023.

Summary of my perl5 (revision 5 version 39 subversion 5) configuration:
  Commit id: 0fc8a78066b687b6a94c809020b00827bef6e265
  Platform:
    osname=openbsd
    osvers=7.4
    archname=OpenBSD.amd64-openbsd-thread-multi
    uname='openbsd cjg-openbsd7 7.4 generic#0 amd64 '
    config_args='-des -Dprefix=/home/cpan/bin/perl -Dscriptdir=/home/cpan/bin/perl/bin -Dusedevel -Duse64bitall -Duseithreads'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=define
    usemultiplicity=define
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
  Compiler:
    cc='cc'
    ccflags ='-pthread -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    optimize='-O2'
    cppflags='-pthread -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    ccversion=''
    gccversion='OpenBSD Clang 13.0.0'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags ='-pthread -Wl,-E  -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/lib/clang/13.0.0/lib /usr/lib /usr/local/lib
    libs=-lpthread -lm -lutil -lc
    perllibs=-lpthread -lm -lutil -lc
    libc=/usr/lib/libc.so.97.1
    so=so
    useshrplib=false
    libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags=' '
    cccdlflags='-DPIC -fPIC '
    lddlflags='-shared -fPIC  -L/usr/local/lib -fstack-protector-strong'


---
@INC for perl 5.39.5:
    /home/cpan/bin/perl/lib/site_perl/5.39.5/OpenBSD.amd64-openbsd-thread-multi
    /home/cpan/bin/perl/lib/site_perl/5.39.5
    /home/cpan/bin/perl/lib/5.39.5/OpenBSD.amd64-openbsd-thread-multi
    /home/cpan/bin/perl/lib/5.39.5

---
Environment for perl 5.39.5:
    HOME=/home/cpan
    LANG (unset)
    LANGUAGE (unset)
    LC_ALL=C
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/cpan/bin/perl/bin:/home/cpan/bin:/home/cpan/bin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin:/usr/local/bin:/usr/local/sbin
    PERL_BADLANG (unset)
    SHELL=/usr/local/bin/bash
```

**Comment by tonycoz** [comment]
This looks like it might be the result of this change in Getopt::Long, from the report:
```
        #   Failed test 'pipe: no stderr'
        #   at /home/cpan/.cpan/build/Code-TidyAll-0.83-0/t/lib/TestFor/Code/TidyAll/Basic.pm line 610.
        #   (in TestFor::Code::TidyAll::Basic->test_cli)
        #                   'Duplicate specification "I=s" for option "i"
        # '
        #           matches '(?^:\S)'
        # Looks like you failed 1 test of 2.
```
From the Getopt::Long changes:
```
* Fix long standing bug that duplicate options were not detected when
  the options differ in case while ignore_case is in effect.
  This will now yield a warning and become a fatal error in a future
  release.
```

**Comment by mauke** [comment]
Yeah, this isn't really tied to blead, only to the version of Getopt::Long used for the test. And it is a genuine bug in Code::TidyAll: https://github.com/houseabsolute/perl-code-tidyall/issues/125

**Comment by mauke** [comment]
Fixed in Code::TidyAll 0.84: https://metacpan.org/release/DROLSKY/Code-TidyAll-0.84/source/Changes
